packages/11ty/_plugins/transforms/outputs/pdf/transform.js

```diff
@@ -35,19 +38,26 @@ module.exports = function(eleventyConfig, collections, content) {
 
     if (mainElement) {
       if (pageIndex !== -1) {
+        const currentPage = collections.pdf[pageIndex].data
+        const {
+          label,
+          parentPage,
+          short_title: shortTitle,
+          title,
+        } = currentPage
         const sectionElement = document.createElement('section')
         sectionElement.innerHTML = mainElement.innerHTML
         for (className of mainElement.classList) {
           sectionElement.classList.add(className)
         }
 
-        const pageLabelDivider = eleventyConfig.globalData.config.params
-        const { label, title } = collections.pdf[pageIndex].data
-
-        // set data attributes for PDF generation
-        sectionElement.dataset.pageTitle = label
-          ? `${label}${pageLabelDivider}${title}`
-          : title
+        // set data attributes for PDF footer
+        const truncatedTitle = shortTitle || truncate(title, 35)
+        const pdfPageTitle = pageTitle({ label, title: truncatedTitle })
+        sectionElement.dataset.pageTitle = pdfPageTitle
+        if (parentPage) {
+          sectionElement.dataset.sectionTitle = parentPage.data.title
+        }
```

Comment:
A ternary could be used here though should not both `pageTitle` and `parentPage.data.title` be truncated? 
```suggestion
        sectionElement.dataset.pageTitle = parentPage
          ? parentPage.data.title
          : pageTitle({ label, title: truncatedTitle })
        }
```

Reply:
Added `footerTitle` method to truncate the titles